src/lib/Libdb/pgsql/pbs_db_utility

```diff
@@ -0,0 +1,1302 @@
+);
+CREATE INDEX nd_idx_cr
+ON pbs.node
+( nd_creattm );
+
+---------------------- QUEUE -------------------------------
+
+/*
+ * Table pbs.queue holds queue information
+ */
+CREATE TABLE pbs.queue (
+    qu_name     TEXT        NOT NULL,
+    qu_type     INTEGER     NOT NULL,
+    qu_creattm  TIMESTAMP   NOT NULL,
+    qu_savetm   TIMESTAMP   NOT NULL,
+    attributes  hstore      NOT NULL default '',
+    CONSTRAINT queue_pk PRIMARY KEY (qu_name)
+);
+CREATE INDEX que_idx_cr
+ON pbs.queue
+( qu_creattm );
+
+
+---------------------- RESERVATION -------------------------
+
+/*
+ * Table pbs.resv holds reservation information
+ */
+CREATE TABLE pbs.resv (
+    ri_resvID       TEXT        NOT NULL,
+    ri_queue        TEXT        NOT NULL,
+    ri_state        INTEGER     NOT NULL,
+    ri_substate     INTEGER     NOT NULL,
+    ri_stime        BIGINT      NOT NULL,
+    ri_etime        BIGINT      NOT NULL,
+    ri_duration     BIGINT      NOT NULL,
+    ri_tactive      INTEGER     NOT NULL,
+    ri_svrflags     INTEGER     NOT NULL,
+    ri_numattr      INTEGER     NOT NULL,
+    ri_resvTag      INTEGER     NOT NULL,
+    ri_un_type      INTEGER     NOT NULL,
+    ri_fromsock     INTEGER     NOT NULL,
+    ri_fromaddr     BIGINT      NOT NULL,
+    ri_savetm       TIMESTAMP   NOT NULL,
+    ri_creattm      TIMESTAMP   NOT NULL,
+    attributes      hstore      NOT NULL default '',
+    CONSTRAINT resv_pk PRIMARY KEY (ri_resvID)
+);
+
+
+---------------------- JOB ---------------------------------
+
+/*
+ * Table pbs.job holds job information
+ */
+CREATE TABLE pbs.job (
+    ji_jobid        TEXT        NOT NULL,
+    ji_state        INTEGER     NOT NULL,
+    ji_substate     INTEGER     NOT NULL,
+    ji_svrflags     INTEGER     NOT NULL,
+    ji_numattr      INTEGER     NOT NULL,
+    ji_ordering     INTEGER     NOT NULL,
+    ji_priority     INTEGER     NOT NULL,
+    ji_stime        BIGINT,
+    ji_endtbdry     BIGINT,
+    ji_queue        TEXT        NOT NULL,
+    ji_destin       TEXT,
+    ji_un_type      INTEGER     NOT NULL,
+    ji_momaddr      BIGINT,
+    ji_momport      INTEGER,
+    ji_exitstat     INTEGER,
+    ji_quetime      BIGINT,
+    ji_rteretry     BIGINT,
+    ji_fromsock     INTEGER,
+    ji_fromaddr     BIGINT,
+    ji_4jid         TEXT,
+    ji_4ash         TEXT,
+    ji_credtype     INTEGER,
+    ji_qrank        BIGINT      NOT NULL,
+    ji_savetm       TIMESTAMP   NOT NULL,
+    ji_creattm      TIMESTAMP   NOT NULL,
+    attributes      hstore      NOT NULL default '',
+    CONSTRAINT jobid_pk PRIMARY KEY (ji_jobid)
+);
+
+CREATE INDEX job_rank_idx
+ON pbs.job
+( ji_qrank );
+
+
+/*
+ * Table pbs.job_scr holds the job script 
+ */
+CREATE TABLE pbs.job_scr (
+    ji_jobid    TEXT       NOT NULL,
+    script      TEXT
+);
+CREATE INDEX job_scr_idx ON pbs.job_scr (ji_jobid);
+
+---------------------- END OF SCHEMA -----------------------
+EOF
+}
+
+upgrade_pbs_schema_from_v1_0_0() {
```

Comment:
We should take a look at which upgrade routines we need. 
1. We need to only support upgrades from last 2 versions of PBS. So, we can eliminate very old schema upgrade functions.
2. We can maintain just 2 files for upgrades from last 2 versions of PBS. upragde_minus_one.sql and upgrade_minus_two.sql, or similarly named. So we keep only two file that pbs_db_utility will refer to and fail in other cases. What say?

Reply:
While we only need to support the last 2 versions of PBS, I don't think we want to make it hard for people from older versions to upgrade to newer versions.  Is it a problem to keep the old upgrade from X to Y around, and then run them in sequence.  If they are on version 1.2, first upgrade them to 1.3, then 1.4, and then finally 1.5?  I agree writing an upgrade function from every old version to the new version isn't worth it, but if we can do it in stages, I don't see the harm in keeping them.

Reply:
I guess we can remove everything below v1_3_0. I moved back the schema upgrade code to its own file as it was before. But removed old schema versions and their code from it. We will have v1_3_0 and v1_4_0.

I am not sure having 2 files just for this would be a good idea. Wouldnt it be a maintenance headache?

Reply:
>  If they are on version 1.2, first upgrade them to 1.3, then 1.4, and then finally 1.5?

Yes, that's what we today. If you are on 1.2, we upgrade in stages till you have 1.5.

Reply:
If that is the case, I don't. see any reason why we don't want to keep the upgrade path.  While we don't support the version of PBS, we do want to entice people to upgrade to one we do support.  If we make it so they have to either upgrade multiple times, or have to install fresh, people will just continue using their old version.

Reply:
> While we only need to support the last 2 versions of PBS, I don't think we want to make it hard for people from older versions to upgrade to newer versions. Is it a problem to keep the old upgrade from X to Y around, and then run them in sequence. If they are on version 1.2, first upgrade them to 1.3, then 1.4, and then finally 1.5? I agree writing an upgrade function from every old version to the new version isn't worth it, but if we can do it in stages, I don't see the harm in keeping them.

I think keeping more code means maintaining more code. We do not test upgrades from older releases, which means aside from this database upgrade part, a direct upgrade is probably anyway broken.

Reply:
I'm not sure I agree there is a maintenance issue here.  Since we upgrade in stages, there is no reason we'd ever change any of these older upgrade schema functions.  As for not testing them, we did.  When we were upgrading from X to Y, we tested it.  I don't think we need to test from old old old X to new Y to assume it will still work.  As long as it has been tested in sections, it should work.  By removing this code, the only way someone can upgrade from a non-supported version of PBS to a supported version is to either upgrade in stages (upgrade from 13 to 19 to 2020) or to do a fresh install.  Customers have active systems and won't want to do a fresh install and lose all their jobs.  What we are saying is that we don't want them to upgrade.  Normally I'd be all for removing code which isn't supported any more.  When it has to do with the desire to move from something unsupported to something that is supported, I disagree.

Reply:
No i am not saying we dont want them to upgrade, i am just saying that they must hop-upgrade. Upgrade to 2 version up in each hop.